compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/IREEGPUAttrs.cpp

```diff
@@ -162,83 +162,96 @@ getUnsupportedMNKShape(MMAIntrinsic intrinsic) {
 
 MMASingleSubgroupLayout getSingleSubgroupLayout(MMAIntrinsic intrinsic,
                                                 MMAFragment fragment) {
+  auto mfmaLhs16xK = [](int64_t k) -> MMASingleSubgroupLayout {
```

Comment:
Wouldn't it be useful if this didn't have a hidden 4 in the implementation that the name `mfmaLhs16xk` doesn't convey?

How about `mfmaLhs16x4xE(int e)` where the caller passes for `e` what is currently `k / 4` ? That would also remove the need for the `assert`.

Reply:
The assert is there because there are some wonderful old intrinsics like `mfma_f32_32x32x1_f32` which perform batched matmul over multiple "blocks"

I figured that just using the K dimension would make it easiest to spot-check that the value is passed is correct

Reply:
The fact that there's a division by 4 is an implementation detail - logically, you're always passing K elements to the MFMA, and the layout tells you how they're distributed